### PR TITLE
ICS004: Remove last packet sent

### PR DESCRIPTION
**Pull request by AdityaSripal**
closes: #1024
closes: #1015
closes: #1023

### CODE REVIEW

File: spec/core/ics-004-channel-and-packet-semantics/README.md [hunk 1]
```diff
@@ -828,7 +828,7 @@ function recvPacket(
     channel = provableStore.get(channelPath(packet.destPort, packet.destChannel))
     abortTransactionUnless(channel !== null)
     counterpartyLastPacketSent = privateStore.get(channelCounterpartyLastPacketSequencePath(packet.destPort, packet.destChannel)
```
Comment (crodriguezvega) [inline spec/core/ics-004-channel-and-packet-semantics/README.md:830]: This line should be removed as well, right?

```suggestion
```

**Review by crodriguezvega — APPROVED** [review]
Nice! 🧹

**Review by sangier — COMMENTED** [review]
If I understand correctly, the core idea here is that we can remove the `lastPacketSent` field because the following statements remain true:

1. 'Any packet sent to the channel with a packet sequence number greater than the `lastPacketSent` will be rejected until the upgrade is complete.'
   - This statement remains valid because, even without the `lastPacketSent` field, the transition to the `FLUSHING` state during the `startFlushUpgradeHandshake` operation effectively blocks the `sendPacket` function. As a result, no additional packets can be sent over this channel during the upgrade process.
   - Considering that we were storing the `lastPacketSent` when we were moving to FLUSHING, the system without `lastPacketSent` achieve the same result. 

2. We still have a way to check if there are pending inflight packets.
   - `pendingInflightPacket` does not depend on `lastPacketSent`. Indeed is designed to monitor the pending in-flight packet verifying the presence/absence of packet commitment given that packet commitments are deleted only when the packet lifecycle is complete. Therefore, if a packet commitment exists on the sender's chain, it indicates that the packet lifecycle is still incomplete.

Am I missing something? If all the above is correct, IMO, removing `lastPacketSent` is safe.

File: spec/core/ics-004-channel-and-packet-semantics/UPGRADES.md [hunk 2]
```diff
@@ -312,10 +301,7 @@ function startFlushUpgradeHandshake(
   // either timeout height or timestamp must be non-zero
```
Comment (sangier) [inline spec/core/ics-004-channel-and-packet-semantics/UPGRADES.md:301]: In the comment al line 287, we should remove "and the las packet sent".

Comment (crodriguezvega) [reply spec/core/ics-004-channel-and-packet-semantics/UPGRADES.md:301]: Yes, good catch!

**Review by damiannolan — APPROVED** [review]
Also closes #1015 as it won't be required after this PR 👍 
Thanks @AdityaSripal ❤️

File: spec/core/ics-004-channel-and-packet-semantics/UPGRADES.md [hunk 3]
```diff
@@ -348,7 +334,6 @@ function openUpgradeHandshake(
```
Comment (sangier) [inline spec/core/ics-004-channel-and-packet-semantics/UPGRADES.md:334]: Typo at line 312: "`openUpgradeHandshake` will open the channel and switch the existing channel parameters to the newly agreed-upon **uprade** channel fields."

Fix: "`openUpgradeHandshake` will open the channel and switch the existing channel parameters to the newly agreed-upon upgrade channel fields."

**Pull request merged**